src/pseudopeople/configuration/validator.py

```diff
@@ -93,105 +101,90 @@ def _get_default_config_node(
 
         error_message = f"Invalid {key_type} '{key}' provided{context}. "
         valid_options_message = f"Valid {key_type}s are {[k for k in default_config]}."
-        raise ValueError(error_message + valid_options_message)
+        raise ConfigurationError(error_message + valid_options_message)
 
 
 def _validate_possible_age_differences(
     noise_type_config: Union[Dict, List],
-    form: str,
-    column: str,
     parameter: str,
-    noise_type: str,
+    invalid_error: str,
 ) -> None:
     """
     Validates the user-provided values for the age-miswriting permutations
     parameter
     """
     if not isinstance(noise_type_config, (Dict, List)):
-        raise TypeError(
-            f"Invalid configuration type provided for '{parameter}'; must a Dict or List. "
-            f"Provided {noise_type_config} of type {type(noise_type_config)} in the "
-            f"configuration for noise type {noise_type} in form {form} and column {column}."
+        raise ConfigurationError(
+            invalid_error + f"'{parameter}' must be a Dict or List. "
+            f"Provided {noise_type_config} of type {type(noise_type_config)}."
         )
-
     for key in noise_type_config:
         if not isinstance(key, int):
-            raise TypeError(
-                "All possible age miswriting differences must be ints. "
-                f"Provided {key} of type {type(key)} in the configuration "
-                f"for form {form} and column {column}."
+            raise ConfigurationError(
+                invalid_error + f"'{parameter}' must be ints. "
```

Comment:
Nit: `f"'{parameter}' must all be ints. "` or `f"'{parameter}' must be a List of ints. "`